docs/reference/snapshot-restore/apis/create-snapshot-api.asciidoc

```diff
@@ -0,0 +1,122 @@
+
+Besides creating a copy of each index, the snapshot process can also store global cluster metadata, including persistent cluster settings and templates. The transient settings and registered snapshot repositories are not stored as part of the snapshot.
+
+[[create-snapshot-api-path-params]]
+==== {api-path-parms-title}
+
+`<repository>`::
+(Required, string)
+Name of the repository to create a snapshot in.
+
+`<snapshot-name>`::
+(Required, string)
+Unique name of the snapshot.
+
+`wait_for_completion`::
+(Optional, boolean)
+Specifies whether the request returns immediately after snapshot initialization (default), or wait for snapshot completion. Defaults to `false`.
++
+NOTE: During snapshot initialization, information about all
+previous snapshots is loaded into memory. In large repositories, this load time can cause this command to take several seconds (or even minutes) to return, even if the `wait_for_completion` parameter is set to `false`.
+
+[[create-snapshot-api-query-params]]
+==== {api-query-parms-title}
+
+`indices`::
+(Optional, string)
+A list of indices to include in the snapshot. This parameter supports <<multi-index,multi index syntax>>, although the options that control the behavior of multi index syntax
+must be supplied in the body of the request, rather than as request parameters.
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=index-ignore-unavailable]
+
+`include_global_state`::
+(Optional, boolean)
+If `false`, this parameter prevents the cluster global state from being stored as part of the snapshot. Defaults to `false`.
++
+If `true`, the snapshot's persistent settings, index templates, ingest
+pipelines, and {ilm-init} policies are restored into the current cluster. This
+overwrites any existing cluster settings, templates, pipelines and {ilm-init}
+policies whose names match those in the snapshot.
++
+IMPORTANT: By default, the entire snapshot will fail if one or more indices participating in the snapshot do not have all primary shards available. You can change this behavior by setting `partial` to `true`.
+
+`metadata`::
+(Optional, string)
+Attaches arbitrary metadata to the snapshot, such as a record of who took the snapshot, why it was taken, or any other useful data.
+
+[[create-snapshot-api-example]]
+==== {api-examples-title}
+
+In the following example, `index_1` and `index_2` are the indices to take snapshots of.
+
+[source,console]
+-----------------------------------
+PUT /_snapshot/my_backup/snapshot_2?wait_for_completion=true
+{
+  "indices": "index_1,index_2",
+  "ignore_unavailable": true,
+  "include_global_state": false,
+  "metadata": {
+    "taken_by": "user123",
+    "taken_because": "backup before upgrading"
+  }
+}
+-----------------------------------
+
+The API returns the following response:
+
+[source,console-result]
+----
+{
+  "accepted" : true
+}
+----
+// TEST[skip:cannot complete subsequent snapshot]
```

Comment:
Snippet tests help ensure our docs are correct. Generally, we don't use `// TEST[skip:...` unless absolutely necessary.  In this case,  I think the skipped test is hiding an incorrect response.

I suggested a valid response below. Here's how you can get it yourself next time:

* Use a command like `./gradlew :docs:integTestRunner --tests "*create-snapshot-api*"` to test the snippets in the file.
* Use `// TESTRESPONSE[s/.../.../]` to selectively replace some properties in the response.

This is covered in more depth in the docs README: https://github.com/elastic/elasticsearch/tree/master/docs

```suggestion
[source,console-result]
----
{
  "snapshot": {
    "snapshot": "snapshot_2",
    "uuid": "vdRctLCxSketdKb54xw67g",
    "version_id": 8000099,
    "version": "8.0.0",
    "indices": [],
    "data_streams": [],
    "include_global_state": false,
    "metadata": {
      "taken_by": "user123",
      "taken_because": "backup before upgrading"
    },
    "state": "SUCCESS",
    "start_time": "2020-06-25T14:00:28.850Z",
    "start_time_in_millis": 1593093628850,
    "end_time": "2020-06-25T14:00:28.850Z",
    "end_time_in_millis": 1593094752018,
    "duration_in_millis": 0,
    "failures": [],
    "shards": {
      "total": 0,
      "failed": 0,
      "successful": 0
    }
  }
}
----
// TESTRESPONSE[s/"uuid": "vdRctLCxSketdKb54xw67g"/"uuid": $body.snapshot.uuid/]
// TESTRESPONSE[s/"start_time": "2020-06-25T14:00:28.850Z"/"start_time": $body.snapshot.start_time/]
// TESTRESPONSE[s/"start_time_in_millis": 1593093628850/"start_time_in_millis": $body.snapshot.start_time_in_millis/]
// TESTRESPONSE[s/"end_time": "2020-06-25T14:00:28.850Z"/"end_time": $body.snapshot.end_time/]
// TESTRESPONSE[s/"end_time_in_millis": 1593094752018/"end_time_in_millis": $body.snapshot.end_time_in_millis/]
// TESTRESPONSE[s/"duration_in_millis": 0/"duration_in_millis": $body.snapshot.duration_in_millis/]
```